### PR TITLE
cmd/snap-confine: build const data structures at compile-time

### PR DESCRIPTION
**Pull request by mardy**
Adding the `static` keyword before initializations of const variables
ensures that such initialization will happen at compile-time. As it can
be seen from this assembly snippet (extracted from the
sc_populate_mount_ns() function), before this change the array of
`sc_mount` structures was computed at run-time:

```as
.LC62:
    .string "/home"
.LC63:
    .string "/root"
.LC64:
    .string "/proc"

[...]

    leaq    .LC62(%rip), %rax
    movq    %rax, 128(%rsp)
    leaq    .LC63(%rip), %rax
    movq    %rax, 160(%rsp)
    leaq    .LC64(%rip), %rax
    movq    %rax, 192(%rsp)
```

whereas, declaring these variables as `static` causes them to be dumped
into the read-only data section, and loaded with a single operation:

```as
.LC115:
    .string "/home"
.LC116:
    .string "/root"
.LC117:
    .string "/proc"

[...]

mounts.3:
    .quad   .LC113
    .zero   24
    .quad   .LC114
    .zero   24
    .quad   .LC115
    .zero   24

[...]

    leaq    mounts.3(%rip), %rax
    movq    %rax, 8(%rsp)
```

### CODE REVIEW

File: cmd/snap-confine/mount-support-nvidia.c [hunk 1]
```diff
@@ -360,7 +360,7 @@ static void sc_mkdir_and_mount_and_glob_files(const char *rootfs_dir,
 static void sc_mount_nvidia_driver_biarch(const char *rootfs_dir, const char **globs, size_t globs_len)
 {
 
-	const char *native_sources[] = {
+	static const char *native_sources[] = {
```
Comment (bboozzoo) [inline cmd/snap-confine/mount-support-nvidia.c:363]: Slightly changes the meaning now. I guess the effect is obtained if this is moved outside just as `const`, but it also would have to be declared with static (again slightly different meaning) as it's local to this *.o. I guess it's better if the arrays are kept close to the place where they are used.

**Review by bboozzoo — APPROVED** [review]
LGTM

File: cmd/snap-confine/mount-support-nvidia.c (hunk 1 above)
Comment (mardy) [reply cmd/snap-confine/mount-support-nvidia.c:363]: Yes, this preserves the existing visibility of the variable, it just changes its storage to be static. It's unfortunate that in C/C++ the `static` keyword as a handful of totally different meanings :-)

**Review by mvo5 — APPROVED** [review]
Thank you

**Pull request merged**